optional-integration-analysis/perform-integration.R

```diff
@@ -0,0 +1,109 @@
+## This script performs integration on processed SCE objects that have been
+## merged in the `merge-sce.R` script
+
+## Set up ----------------------------------------------------------------------
+
+# Load libraries
+library(optparse)
+library(dplyr)
+library(SingleCellExperiment)
+library(scpcaTools)
+
+# Declare command line options
+option_list <- list(
+  make_option(
+    opt_str = c("--merged_sce_file"),
+    type = "character",
+    help = "Path to the RDS file containing the merged SCE object"
+  ),
+  make_option(
+    opt_str = c("--integration_method"),
+    type = "character",
+    default = "fastMNN, harmony",
+    help = "The integration method(s) to use when performing integration; 
+    default is 'fastMNN, harmony'"
+  ),
+  make_option(
+    opt_str = c("-o", "--output_sce_file"),
+    type = "character",
+    help = "Path to output RDS file containing integrated object, must end in .rds"
+  ),
+  make_option(
+    c("--project_root"),
+    type = "character",
+    default = NULL,
+    help = "Path to the root directory for the R project and where the `utils` folder lives."
+  )
+)
+
+# Read the arguments passed
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# If project root is not provided use here::here()
+if(is.null(opt$project_root)){
+  project_root <- here::here()
+} else {
+  project_root <- opt$project_root
+}
+
+# Source in set up functions
+source(file.path(project_root, "utils", "setup-functions.R"))
+
+# Check R version
+check_r_version()
+
+# Set up renv
+setup_renv(project_filepath = project_root)
+
+# Check that file extension for output file is correct
+if(!(stringr::str_ends(opt$output_sce_file, ".rds"))){
+  stop("output file name must end in .rds")
+}
+
+# Check that merged SCE file exists
+if(!file.exists(opt$merged_sce_file)){
+  stop(paste(opt$merged_sce_file, "does not exist."))
+}
+
+# Read in merged SCE -----------------------------------------------------------
+
+merged_sce <- readr::read_rds(opt$merged_sce_file)
+
+# Integrate SCEs ---------------------------------------------------------------
+
+# Split up string of integration methods
+integration_methods <-  stringr::str_split(opt$integration_method, ",") %>%
+  unlist() %>%
+  stringr::str_trim()
+
+# Check that we support the provided integration method(s)
+for (integration_method in integration_methods) {
+  if (!integration_method %in% c("fastMNN", "harmony")) {
+    stop("--integration_method must be either fastMNN, harmony, or both (comma-separated)")
+  }
+}
```

Comment:
```suggestion
if (!all(integration_method %in% c("fastMNN", "harmony"))) {
    stop("--integration_method must be either fastMNN, harmony, or both (comma-separated)")
  }
```
You should be able to check these together with `all`